maps/away/russianderelict/russianderelict.dm

```diff
@@ -252,3 +259,34 @@
 
 /obj/effect/floor_decal/urist/russian
 	icon_state = "derelict1"
+
+//contract
+
+/datum/contract/russianderelict
+	name = "Derelict Repower Contract"
+	desc = "Restore power to this station by setting up the singularity."
+	rep_points = 3
+	money = 10000
+	amount = 1
+
+/obj/machinery/the_singularitygen/russianderelict
+	var/contract_enabled = FALSE
+
+/obj/machinery/the_singularitygen/russianderelict/Initialize()
+	.=..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/the_singularitygen/russianderelict/LateInitialize()
+	if(!contract_enabled && GLOB.using_map.use_overmap)
+		contract_enabled = TRUE
```

Comment:
this whole bit really isn't necessary. the reason i did it this way for the destroyed_colony one is because the radio beacon spawns an object on the overmap when the distress signal is activated, which causes runtimes when the map is loaded without an overmap, such as in awaymap testing. The contract_enabled var was because deactivate() could be called multiple times on the beacon, so it prevents it from checking for contract completion every time it's deactivated. However, the singulogen qdels itself after it is activated, so that isn't an issue here either.

Reply:
Done along with slight desc expansion